frontend/src/metabase/actions/containers/ActionCreator/ActionCreatorView.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useCallback, useState } from "react";
+import { t } from "ttag";
+import type { ReactNode } from "react";
+import Button from "metabase/core/components/Button";
+import Modal from "metabase/components/Modal";
+import { ActionFormSettings } from "metabase-types/api";
+import ActionCreatorHeader from "metabase/actions/containers/ActionCreator/ActionCreatorHeader";
+import QueryActionEditor from "metabase/actions/containers/ActionCreator/QueryActionEditor";
+import FormCreator from "metabase/actions/containers/ActionCreator/FormCreator";
+import {
+  DataReferenceTriggerButton,
+  DataReferenceInline,
+} from "metabase/actions/containers/ActionCreator/InlineDataReference";
+import {
+  ActionCreatorBodyContainer,
+  EditorContainer,
+  ModalRoot,
+  ModalActions,
+  ModalLeft,
+} from "metabase/actions/containers/ActionCreator/ActionCreator.styled";
+
+import type NativeQuery from "metabase-lib/queries/NativeQuery";
+
+import Question from "metabase-lib/Question";
+
+import type { SideView } from "./types";
+
+interface ActionCreatorProps {
+  isNew: boolean;
+  canSave: boolean;
+
+  question: Question;
+
+  onChangeQuestionQuery: (query: NativeQuery) => void;
+  onChangeName: (name: string) => void;
+  onCloseModal?: () => void;
+  onChangeFormSettings: (formSettings: ActionFormSettings) => void;
+  onClickSave: () => void;
+  onClickExample: () => void;
+}
+
+const DEFAULT_SIDE_VIEW: SideView = "actionForm";
+
+export default function ActionCreatorView({
+  isNew,
+  canSave,
+  question,
+  onChangeQuestionQuery,
+  onChangeName,
+  onCloseModal,
+  onChangeFormSettings,
+  onClickSave,
+  onClickExample,
+}: ActionCreatorProps) {
+  const [activeSideView, setActiveSideView] =
```

Comment:
@kulyk All state regarding sidebar view are moved from `<ActionCreator />` to here.